mithril-stm/src/multi_sig.rs

```diff
@@ -287,10 +291,14 @@ impl From<&SigningKey> for ProofOfPossession {
 impl Signature {
     /// Verify a signature against a verification key.
     pub fn verify(&self, msg: &[u8], mvk: &VerificationKey) -> Result<(), MultiSignatureError> {
-        blst_err_to_mithril(
-            self.0.verify(false, msg, &[], &[], &mvk.0, false),
-            Some(*self),
-        )
+        match self.0.validate(true) {
+            Ok(_) => blst_err_to_mithril(
+                self.0.verify(false, msg, &[], &[], &mvk.0, false),
+                Some(*self),
+                None,
+            ),
+            Err(e) => blst_err_to_mithril(e, Some(*self), None),
+        }
```

Comment:
Maybe this would be more readable:
```suggestion
        blst_err_to_mithril(
            self.0.validate(true).map_or_else(
                |e| e,
                |_| self.0.verify(false, msg, &[], &[], &mvk.0, false),
            ),
            Some(*self),
            None,
        )
```